framework/src/main/java/org/checkerframework/framework/util/AnnotatedTypes.java

```diff
@@ -390,8 +450,35 @@ private static AnnotatedTypeMirror asMemberOfImpl(
                         member);
             case INTERSECTION:
             case UNION:
+            case DECLARED:
+                // return substituteTypeVariables(types, atypeFactory, of, member, memberType);
+                return memberType;
+            default:
+                throw new BugInCF("asMemberOf called on unexpected type.\nt: " + of);
+        }
+    }
+
+    private static AnnotatedTypeMirror asMemberOfImpl(
```

Comment:
Again, javadoc to make clear what the method does.
Why is this extra method needed? Instead, couldn't you split up the original method into two separate methods?